pygsti/evotypes/evotype.py

```diff
@@ -50,10 +51,46 @@ class Evotype(object):
     }
 
     @classmethod
-    def cast(cls, obj, default_prefer_dense_reps=False):
+    def cast(cls, obj, default_prefer_dense_reps=None, state_space=None):
```

Comment:
I see that this API change affects many files. In an effort to make sure we've caught all instances of calling this function, consider temporarily modifying this function to raise a runtime error if state_space is None and then running tests under that modification. (This is just a suggestion. Feel free to mark this comment as resolved and ignore.)

Reply:
I'm pretty sure I caught all of the spots where this API change is relevant to forward simulation performance. The old API will still work as-is for any places that might have been missed. The consequences of missing this somewhere should only really be suboptimal simulation performance.

Read: I probably won't do that, but the suggestion is well-taken and objectively good advice.